exonum/src/api/public/blockchain_explorer.rs

```diff
@@ -50,34 +71,66 @@ impl ExplorerApi {
         Ok(explorer.blocks_range(count, from, skip_empty_blocks))
     }
 
-    fn get_block(&self, height: Height) -> Result<Option<BlockInfo>, ApiError> {
+    fn block(&self, height: Height) -> Option<BlockInfo> {
         let explorer = BlockchainExplorer::new(&self.blockchain);
-        Ok(explorer.block_info(height))
+        explorer.block_info(height)
+    }
+
+    fn transaction_info(&self, hash: &Hash) -> Result<TransactionInfo, ApiError> {
+        if let Some(tx) = self.pool.read().expect("Uanble to read pool").get(hash) {
+            Ok(TransactionInfo::InPool(PoolTransactionInfo {
+                content: tx.serialize_field().map_err(ApiError::InternalError)?,
+            }))
+        } else if let Some(tx_info) = BlockchainExplorer::new(&self.blockchain).tx_info(hash)? {
+            Ok(TransactionInfo::Committed(CommittedTransactionInfo{tx_info}))
+        } else {
+            Ok(TransactionInfo::Unknown)
+        }
     }
 }
 
 impl Api for ExplorerApi {
     fn wire(&self, router: &mut Router) {
+        set_blocks_response(self.clone(), router);
+        set_block_response(self.clone(), router);
+        set_transaction_info_response(self.clone(), router);
+    }
+}
 
-        let self_ = self.clone();
-        let blocks = move |req: &mut Request| -> IronResult<Response> {
-            let count: u64 = self_.required_param(req, "count")?;
-            let latest: Option<u64> = self_.optional_param(req, "latest")?;
-            let skip_empty_blocks: bool = self_
-                .optional_param(req, "skip_empty_blocks")?
-                .unwrap_or(false);
-            let info = self_.get_blocks(count, latest, skip_empty_blocks)?;
-            self_.ok_response(&::serde_json::to_value(info).unwrap())
-        };
+fn set_blocks_response(api: ExplorerApi, router: &mut Router) {
```

Comment:
I'm not sure I like this free function syntax. IMO, it would make sense to make these functions consuming methods of `ExplorerApi` and invoke them like `self.clone().handle_blocks(router)`.